azure/client.go

```diff
@@ -36,8 +37,14 @@ func NewAnonymousClient() Client {
 
 //NewClientFromPublishSettingsFile creates a new azure.Client and imports the publish
 //settings from the specified file path.
-func NewClientFromPublishSettingsFile(publishSettingsFilePath string) (Client, error) {
-	client := Client{}
+func NewClientFromPublishSettingsFile(publishSettingsFilePath string, serviceManagementURL ...string) (Client, error) {
+	var client Client
+	if len(serviceManagementURL) > 0 {
```

Comment:
So my use of a variadic parameter was to create an "optional" parameter. You'd never pass multiple values to this. Wasn't sure if it was best practice to do this or create new functions altogether